provider/pkg/provider/provider.go

```diff
@@ -1358,9 +1358,8 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 	// needs to be `DeleteBeforeReplace`'d. If the resource is marked `DeleteBeforeReplace`, then
 	// `Create` will allocate it a new name later.
 	if len(oldInputs.Object) > 0 {
-		// NOTE: If old inputs exist, they have a name, either provided by the user or filled in with a
+		// NOTE: If old inputs exist, they MAY have a name, either provided by the user or filled in with a
 		// previous run of `Check`.
-		contract.Assertf(oldInputs.GetName() != "", "expected object name to be nonempty: %v", oldInputs)
```

Comment:
Should double-check to ensure that removing this assertion doesn't have implications elsewhere in the provider. The code was written with the assumption that the name was always present, although the actual update value didn't always match the preview value in the case of auto-naming, IIRC.